sdk/test/opentelemetry/sdk/trace/export/simple_span_processor_test.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+def stub_span_builder(sampled: false)
+  ctx = OpenTelemetry::Trace::SpanContext.new
+  span = OpenTelemetry::Trace::Span.new(span_context: ctx)
+  def span.to_span_data; end
+  if sampled
+    def span.recording_events?
+      true
+    end
+  end
```

Comment:
Why this vs.
```ruby
def span.recording_events?
  recording
end
```

Reply:
I'm not sure I understand what this is trying to do. Don't we just want to return the `recording` value here?

Reply:
Yes, I was following the test code too literally when I wrote this, and clearly didn't take a second glance for sanity. Just returning the value of `recording` is fine.

Reply:
@rdooley ^

Reply:
It's doing that, but because the comment is on the shared end line the comment pane on github isn't getting updated i think 
https://github.com/open-telemetry/opentelemetry-ruby/pull/98/files#diff-664f94ed9cc21f7f4fe36d14d27ddc07R14
which was introduced in https://github.com/open-telemetry/opentelemetry-ruby/pull/98/commits/e990865865b3a0b0afe9c69117f2ce445c8d516c

Reply:
<img width="300" alt="opentelemetry-ruby 2019-10-01 14-11-03" src="https://user-images.githubusercontent.com/1551119/65988150-4e636280-e455-11e9-99ec-7c987605cfcc.png">

^ is what I see. This should be:
```ruby
def span.recording_events?
  recording
end
```

Reply:
If I do not wrap in the lambda I get two test errors
```
Error:
OpenTelemetry::SDK::Trace::Export::SimpleSpanProcessor#test_0003_forwards recorded spans from #on_finish:
MockExpectationError: expected export(Array) => 0
    PATH_REDACTED/opentelemetry-ruby/sdk/test/opentelemetry/sdk/trace/export/simple_span_processor_test.rb:52:in `block (2 levels) in <top (required)>'
```
and
```
Error:
OpenTelemetry::SDK::Trace::Export::SimpleSpanProcessor#test_0006_catches and logs exporter exceptions in #on_finish:
MockExpectationError: mocked method :error called with unexpected arguments ["unexpected error in span.on_finish - undefined local variable or method `recording' for #<OpenTelemetry::Trace::Span:0x00007fd08a961ee8>\nDid you mean?  recording_events?"]
    PATH_REDACTED/opentelemetry-ruby/sdk/lib/opentelemetry/sdk/trace/export/simple_span_processor.rb:54:in `rescue in on_finish'
    PATH_REDACTED/opentelemetry-ruby/sdk/lib/opentelemetry/sdk/trace/export/simple_span_processor.rb:49:in `on_finish'
    PATH_REDACTED/opentelemetry-ruby/sdk/test/opentelemetry/sdk/trace/export/simple_span_processor_test.rb:91:in `block (3 levels) in <top (required)>'
    PATH_REDACTED/opentelemetry-ruby/sdk/test/opentelemetry/sdk/trace/export/simple_span_processor_test.rb:90:in `block (2 levels) in <top (required)>'
```

Reply:
I played around with this. You need to define the method differently, in order to access `recording`:
```ruby
span.define_singleton_method(:recording_events?) { recording }
```